src/main/java/sirius/kernel/xml/StructuredNode.java

```diff
@@ -143,6 +149,65 @@ public Node getNode() {
         return resultList;
     }
 
+    /**
+     * Returns a map of all attributes of this DOM node with their names as keys.
+     *
+     * @return a list containing all attributes of this node. If no attributes exist, an empty list will be returned.
+     */
+    @Nonnull
+    public ImmutableMap<String, StructuredNode> getAttributes() {
+        if (nodeAttributes == null) {
+            compileAttributes();
+        }
+        return ImmutableMap.copyOf(nodeAttributes);
+    }
+
+    private void compileAttributes() {
+        Map<String, StructuredNode> attributes = new HashMap<>();
+        NamedNodeMap result = node.getAttributes();
+        if (result != null) {
+            for (int i = 0; i < result.getLength(); i++) {
+                StructuredNode structuredNode = new StructuredNode(result.item(i));
+                attributes.put(structuredNode.getNodeName(), structuredNode);
+            }
+        }
+        nodeAttributes = attributes;
+    }
+
+    /**
+     * Returns the attribute with the given name.
+     *
+     * @return the attribute as a {@link StructuredNode} if an attribute exists for the given name, <tt>null</tt> otherwise
+     */
+    @Nullable
+    public StructuredNode getAttribute(String name) {
+        return getAttributes().get(name);
+    }
+
+    /**
+     * Returns the value of the  attribute with the given name.
+     *
+     * @return the value of the attribute as a {@link String} if an attribute exists for the given name, <tt>""</tt> otherwise
+     */
+    @Nonnull
+    public String getAttributeValue(String name) {
+        StructuredNode attribute = getAttribute(name);
+        if (attribute != null) {
+            return attribute.getNode().getNodeValue();
+        }
+        return "";
```

Comment:
Or should null be returned in this case?

Reply:
Maybe an Optional?

Reply:
Optional or simply just use Value as stated above...